test/query.jl

```diff
@@ -286,16 +286,20 @@ end
 
 end
 @testset "Multiple Magic bytes" begin
-    q = query(joinpath(file_dir, "magic1.tiff"))
-    @test typeof(q) == File{format"TIFF"}
-    q = query(joinpath(file_dir, "magic2.tiff"))
-    @test typeof(q) == File{format"TIFF"}
+    q = query(joinpath(file_dir, "ffc.pcx"))
+    @test typeof(q) == File{format"PCX"}
     open(q) do io
         @test position(io) == 0
         skipmagic(io)
-        @test position(io) == 4
```

Comment:
Now that each function has its own type, we could change how `skipmagic` works:
```julia
skipmagic(io, ::typeof(detect_ometiff)) = seek(io, 4)
skipmagic(io, ::typeof(detect_noometiff)) = seek(io, 4)
```
and then I think the original test would pass?